packages/dds/tree/src/test/edit-manager/editManager.spec.ts

```diff
@@ -200,373 +201,265 @@ function asDelta(intentions: number[]): Delta.Root {
-        assert.deepEqual(manager.addSequencedChange(c6), asDelta([6]));
-        assert.deepEqual(anchors.intentions, [1, 2, 3, 4, 5, 6]);
-        checkChangeList(manager, [1, 2, 3, 4, 5, 6]);
+        /** Ordered list of local commits that have not yet been sequenced (i.e., `pushed - acked`) */
+        const localCommits: TestCommit[] = [];
+        /** Ordered list of intentions that the manager has been made aware of (i.e., `pushed ⋃ pulled`). */
+        let knownToLocal: number[] = [];
+        /** Ordered list of intentions that have been sequenced (i.e., `acked ⋃ pulled`) */
+        const trunk: number[] = [];
+        /** The sequence number of the most recent sequenced commit that the manager is aware of */
+        let localRef: number = 0;
+        for (const step of steps) {
+            const { seq, type } = step;
+            switch (type) {
+                case "Push": {
+                    const changeset = TestChangeRebaser.mintChangeset(knownToLocal, seq);
+                    localCommits.push({
+                        sessionId: localSessionId,
+                        seqNumber: brand(seq),
+                        refNumber: brand(localRef),
+                        changeset,
+                    });
+                    knownToLocal.push(seq);
+                    // Local changes should always lead to a delta that is equivalent to the local change.
+                    assert.deepEqual(manager.addLocalChange(changeset), asDelta([seq]));
+                    break;
+                }
+                case "Ack": {
+                    const commit = localCommits.shift();
+                    if (commit === undefined) {
+                        fail("Invalid test scenario: no local commit to acknowledge");
+                    }
+                    if (commit.seqNumber !== seq) {
+                        fail("Invalid test scenario: acknowledged commit does not mach oldest local change");
+                    }
+                    // Acknowledged (i.e., sequenced) local changes should always lead to an empty delta.
+                    assert.deepEqual(manager.addSequencedChange(commit), Delta.empty);
+                    trunk.push(seq);
+                    localRef = seq;
+                    break;
+                }
+                case "Pull": {
+                    /** Filter that includes changes that were on the trunk of the issuer of this commit. */
+                    const peerTrunkChangesFilter = (s: UnitTestScenarioStep) =>
+                        s.seq <= step.ref;
+                    /** Filter that includes changes that were local to the issuer of this commit. */
+                    const peerLocalChangesFilter = (s: UnitTestScenarioStep) =>
+                        s.seq > step.ref && s.seq < step.seq && s.type === "Pull" && s.from === step.from;
+                    /** Changes that were known to the peer at the time it authored this commit. */
```

Comment:
I'm curious why these use doc comment syntax.

Reply:
I though a doc comment would be more appropriate since I'm documenting what an entity is as opposed to documenting what the code is doing.
It has the added perk of showing the doc comment when hovering over the symbol wherever it is used, but that's not really important here.

Reply:
If you're going to do that, why are the comments below this not also doc comments?